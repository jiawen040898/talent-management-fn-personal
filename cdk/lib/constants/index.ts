export * from './resource-tag.enum';
