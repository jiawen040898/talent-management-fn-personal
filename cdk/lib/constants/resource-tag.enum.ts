export enum ResourceTag {
    SNS = 'AWS::SNS::Topic',
    SQS = 'AWS::SQS::Queue',
    IAM = 'AWS::IAM::Role',
    LAMBDA = 'AWS::Lambda::Function',
    S3 = 'AWS::S3::Bucket',
    SSM = 'AWS::SSM::Parameter',
    STATE_MACHINE = 'AWS::StepFunctions::StateMachine',
}
