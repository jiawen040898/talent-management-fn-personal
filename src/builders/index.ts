export * from './base-query.builder';
export * from './employee-task-query.builder';
export * from './feedback-cycle-reviewee-query.builder';
export * from './feedback-cycle-reviewer-query.builder';
export * from './learning-goal-query.builder';
