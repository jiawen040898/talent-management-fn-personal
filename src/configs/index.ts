export * from './aws.config';
export * from './gcp.config';
export * from './recommendation.config';
