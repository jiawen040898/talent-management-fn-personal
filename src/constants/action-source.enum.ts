export enum ActionSource {
    LEARNER = 'learner',
    MANAGER = 'manager',
    ADMIN = 'admin',
}
