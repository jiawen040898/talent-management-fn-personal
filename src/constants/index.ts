export * from './action-source.enum';
export * from './big-queries.enum';
export * from './cache-store.enum';
export * from './company-module.enum';
export * from './company-status.enum';
export * from './due-type.enum';
export * from './employee-score.enum';
export * from './employee-task.enum';
export * from './event-type.enum';
export * from './feedback-recommendation.enum';
export * from './fit-score-recipe.enum';
export * from './person-score.enum';
export * from './psychology-api-errors.enum';
export * from './questionnaire-framework-type.enum';
export * from './sqs-event-type.enum';
export * from './unleash-errors.enum';
