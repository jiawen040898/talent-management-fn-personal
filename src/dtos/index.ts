export * from './company.dto';
export * from './cron-schedule-message.dto';
export * from './employee.dto';
export * from './employee-account-created-event.dto';
export * from './employee-assessment-score-event.dto';
export * from './employee-fit-score-event.dto';
export * from './feedback-cycle.dto';
export * from './fit-score-recipe.dto';
export * from './manager-subordinate-aggregated-report-data.dto';
