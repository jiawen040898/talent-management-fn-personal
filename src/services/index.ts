export * from './cognitive-score.service';
export * from './company.service';
export * from './domain-event-router.service';
export * from './employee-daily-reminder.service';
export * from './employee-management.service';
export * from './employee-progress-report.service';
export * from './employee-score.service';
export * from './feedback-cycle-action.service';
export * from './feedback-cycle-dashboard-provision.service';
export * from './feedback-cycle-reminder.service';
export * from './feedback-cycle-reviewee-recommendation.service';
export * from './fit-score.service';
export * from './personality-score.service';
export * from './psychology.service';
export * from './recommendation.service';
