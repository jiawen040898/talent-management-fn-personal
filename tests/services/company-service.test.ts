import { cacheService, secretService } from '@pulsifi/fn';
import { authService } from '@pulsifi/fn/services/auth';
import axios from 'axios';

import { CompanyModule, CompanyStatus } from '../../src/constants';
import { CompanyLookupDto } from '../../src/dtos';
import { CompanyService } from '../../src/services';
import { testData } from './fixtures/test-data';

describe('CompanyService', () => {
    let sut: CompanyService;

    const spyCacheGet = jest
        .spyOn(cacheService, 'get')
        .mockImplementation(() => {
            return Promise.resolve(JSON.stringify(jest.fn()));
        });

    jest.spyOn(authService, 'getMachineToken').mockImplementation(() => {
        return Promise.resolve(
            testData.mockMachineTokenResponse.data.access_token,
        );
    });

    jest.spyOn(axios, 'get').mockImplementation(() => {
        return Promise.resolve(testData.mockCompanyLookupList);
    });

    jest.spyOn(secretService, 'getSecret').mockImplementation(() => {
        return Promise.resolve(testData.mockAuth0CredentialSecretResponse);
    });

    beforeAll(async () => {
        sut = new CompanyService();
    });

    describe('getCompanyLookup', () => {
        describe('should get data from identity endpoint', () => {
            let actual: CompanyLookupDto[];

            beforeAll(async () => {
                // Arrange
                spyCacheGet.mockImplementationOnce(() => {
                    return Promise.resolve(null);
                });

                // Act
                actual = await sut.getCompanyLookup();
            });

            it('should return company lookup data', async () => {
                // Assert
                expect(actual).toMatchSnapshot();
            });
        });

        describe('should get company lookup data from cache', () => {
            beforeAll(async () => {
                // Arrange
                spyCacheGet.mockClear();

                // Act
                await sut.getCompanyLookup();
            });

            it('should return company lookup data', async () => {
                // Assert
                expect(spyCacheGet).toHaveBeenCalledTimes(1);
                expect(spyCacheGet.mock.calls).toMatchSnapshot();
            });
        });
    });

    describe('getActiveCompaniesSubscribedToLearningAndByTimezone', () => {
        describe('should filter and return companies correctly', () => {
            let actual: CompanyLookupDto[];
            const timezoneList = ['Japan'];

            beforeAll(async () => {
                // Act
                actual =
                    await sut.getActiveCompaniesSubscribedToLearningAndByTimezone(
                        timezoneList,
                    );
            });

            it('should filter and return companies data', async () => {
                // Assert
                expect({ actual }).toMatchSnapshot({
                    actual: new Array(actual.length).fill({
                        localDate: expect.any(String),
                    }),
                });

                actual.forEach((item) => {
                    expect(item.is_deleted).toBeFalsy();
                    expect(item.status).toEqual(CompanyStatus.ACTIVE);
                    expect(
                        timezoneList.some((tz) => item.timezone.includes(tz)),
                    ).toBe(true);
                    expect(
                        item.products.some((prd) =>
                            CompanyModule.TALENT_LEARNING.includes(prd.module),
                        ),
                    ).toBe(true);
                });
            });
        });
    });
});
