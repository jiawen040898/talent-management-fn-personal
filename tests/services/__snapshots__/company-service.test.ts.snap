// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyService getActiveCompaniesSubscribedToLearningAndByTimezone should filter and return companies correctly should filter and return companies data 1`] = `
{
  "actual": [
    {
      "id": 5,
      "is_deleted": false,
      "localDate": Any<String>,
      "locales": [
        {
          "is_default": true,
          "locale": "en-US",
        },
      ],
      "name": "Bitcoin Incorporation",
      "products": [
        {
          "module": "talent_management",
        },
        {
          "module": "talent_acquisition",
        },
        {
          "module": "talent_learning",
        },
      ],
      "status": "active",
      "timezone": "Japan",
    },
  ],
}
`;

exports[`CompanyService getCompanyLookup should get company lookup data from cache should return company lookup data 1`] = `
[
  [
    "COMPANY_LOOKUP",
  ],
]
`;

exports[`CompanyService getCompanyLookup should get data from identity endpoint should return company lookup data 1`] = `
[
  {
    "id": 5,
    "is_deleted": false,
    "locales": [
      {
        "is_default": true,
        "locale": "en-US",
      },
    ],
    "name": "Bitcoin Incorporation",
    "products": [
      {
        "module": "talent_management",
      },
      {
        "module": "talent_acquisition",
      },
      {
        "module": "talent_learning",
      },
    ],
    "status": "active",
    "timezone": "Japan",
  },
  {
    "id": 1471,
    "is_deleted": true,
    "locales": [
      {
        "is_default": true,
        "locale": "en-US",
      },
    ],
    "name": "Blick, Nikolaus and Kuhn updated",
    "products": [
      {
        "module": "talent_acquisition",
      },
      {
        "module": "talent_management",
      },
    ],
    "status": "disabled",
    "timezone": "Asia/Kuala_Lumpur",
  },
  {
    "id": 352,
    "is_deleted": false,
    "locales": [
      {
        "is_default": true,
        "locale": "en-US",
      },
    ],
    "name": "Herman Thiel",
    "products": [
      {
        "module": "talent_acquisition",
      },
      {
        "module": "talent_management",
      },
    ],
    "status": "disabled",
    "timezone": "Asia/Kuala_Lumpur",
  },
]
`;
